### PR TITLE
Search

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Features added 
- Search 

### Other changes
-  Added loader
- Readme
- Fix responsiveness

### CODE REVIEW

File: src/components/post-cta-bar/PostCtaBar.jsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ const PostCtaBar = ({ likes, postId, commentCount, token, currentUser, post_Id }
   const isLiked = likes?.likedBy?.find(user => user.username === currentUser.username);
 
   const copyUrlTOClipboard = async () => {
-    await navigator.clipboard.writeText(`https://orion-space.netlify.app/${postId}`);
+    await navigator.clipboard.writeText(`https://orion-space.netlify.app/post/${postId}`);
```
Comment (vijaytembugade) [inline src/components/post-cta-bar/PostCtaBar.jsx:28]: This is something new I got to know.

File: src/features/user/userSlice.js [hunk 2]
```diff
@@ -85,10 +88,28 @@ export const userSlice = createSlice({
...
       state.uploadingPhoto = true;
     },
+    searchUser: (state, { payload }) => {
+      state.searchTerm = payload;
+      state.foundUsers = state.users.filter(
+        user =>
+          user.username.toLowerCase().includes(payload.trim().toLowerCase()) ||
+          user.firstName.toLowerCase().includes(payload.trim().toLowerCase()) ||
+          user.lastName.toLowerCase().includes(payload.trim().toLowerCase())
+      );
```
Comment (vijaytembugade) [inline src/features/user/userSlice.js:98]: Search functionality is amazing. Works smooth.

File: src/utils/debounce.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  let timerId;
+  return function (...arg) {
+    clearTimeout(timerId);
+    timerId = setTimeout(() => {
+      fn(...arg);
+    }, delay);
+  };
+};
+
+export { debounce };
```
Comment (vijaytembugade) [inline src/utils/debounce.js:11]: Good to see debouncing code

File: src/utils/getPostTime.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+const getPostTime = postDate => {
```
Comment (vijaytembugade) [inline src/utils/getPostTime.js:1]: Its good that you did is manually, although, using react-moment or date-fns would be preferred.

**Review by vijaytembugade — APPROVED** [review]
Good project. Works smooth.
UI is clean and neat. 
Responsiveness is also impressive. 
Good to get into final merge

File: src/components/post-cta-bar/PostCtaBar.jsx (hunk 1 above)
Comment (msdeshmukh009) [reply src/components/post-cta-bar/PostCtaBar.jsx:28]: It is used for copying links to the clipboard.

File: src/features/user/userSlice.js (hunk 2 above)
Comment (msdeshmukh009) [reply src/features/user/userSlice.js:98]: thanks

File: src/utils/getPostTime.js (hunk 4 above)
Comment (msdeshmukh009) [reply src/utils/getPostTime.js:1]: I haven't explored react-moment yet. Will look into it sometime later.

**Review by manojasarsa — APPROVED** [review]
The search feature is working as expected. 
The code looks clean.
Great job!

**Review by AbhishekkGautam — APPROVED** [review]
Everything looks good.
Code is clean & well structured.

**Pull request merged**